### PR TITLE
CV2-3916 display archived action in item history

### PR DESCRIPTION
**Pull request by melsawy**
## Description

Display archived actions in item history dialog.

References: CV2-3916

## Type of change

- [ ] Performance improvement and/or refactoring (non-breaking change that keeps existing functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Security mitigation or enhancement
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Automated test (add or update automated tests)

## How has this been tested?

1. Create a new item.
2. Apply the following actions to item in step 1
    - Mark as Spam
    - Send to Trash
    - Restore from Trash
    - Not Spam
3. Access item in step 1 and open item history dialog
Result: Should see all actions in step 2 

## Things to pay attention to during code review

Please describe parts of the change that require extra attention during code review, for example:

- File FFFF, line LL: This refactoring does this and this. Is it consistent with how it’s implemented elsewhere?
- Etc.

## Checklist

- [x] I have performed a self-review of my own code
- [ ] I've made sure my branch is runnable and given good testing steps in the PR description
- [ ] I considered secure coding practices when writing this code. Any security concerns are noted above.
- [ ] I have commented my code in hard-to-understand areas, if any
- [ ] I have made needed changes to the README
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] If I implemented any new components, they are self-contained, their `propTypes` are declared and they use React Hooks and, if data-fetching is required, they use Relay Modern with fragment containers
- [ ] If my components involve user interaction - specifically button, text fields, or other inputs - I have added a [BEM-like class name](https://meedan.atlassian.net/wiki/spaces/ENG/pages/1327628289/Naming+conventions+for+interactive+elements) to the element that is interacted with
- [ ] To the best of my knowledge, any new styles are applied according to the design system
- [ ] If I added a new external dependency, I included a rationale for doing so and an estimate of the change in bundle size (e.g., checked in https://bundlephobia.com/)

### CODE REVIEW

File: src/app/components/annotations/Annotation.js [hunk 1]
```diff
@@ -628,37 +629,95 @@ class Annotation extends Component {
...
+      const itemChanges = safelyParseJSON(activity.object_changes_json);
+      if (itemChanges.source_id) {
+        const meta = safelyParseJSON(activity.meta);
+        if (meta && meta.source_name) {
+          if (itemChanges.source_id[0] === null) {
+            contentTemplate = (
+              <span>
+                <FormattedMessage
+                  id="annotation.addSource"
+                  defaultMessage="Source {name} add by {author}"
```
Comment (caiosba) [inline src/app/components/annotations/Annotation.js:641]: *added

File: src/app/components/annotations/Annotation.js [hunk 2]
```diff
@@ -628,37 +629,95 @@ class Annotation extends Component {
...
-        } else {
+        } else if (itemChanges.archived[1] === CheckArchivedFlags.TRASHED) {
           contentTemplate = (
             <span>
               <FormattedMessage
-                id="annotation.updateSource"
-                defaultMessage="Source {name} updated by {author}"
-                description="Log entry indicating an item status has been changed"
+                id="annotation.markAsSpam"
+                defaultMessage="Send to Trash by {author}"
```
Comment (caiosba) [inline src/app/components/annotations/Annotation.js:685]: *Sent

File: src/app/components/annotations/Annotation.js [hunk 3]
```diff
@@ -628,37 +629,95 @@ class Annotation extends Component {
...
+                  }}
+                />
+              </span>
+            );
+          } else if (itemChanges.archived[0] === CheckArchivedFlags.TRASHED) {
+            contentTemplate = (
+              <span>
+                <FormattedMessage
+                  id="annotation.restoreFromTrash"
+                  defaultMessage="Restore from Trash by {author}"
```
Comment (caiosba) [inline src/app/components/annotations/Annotation.js:712]: *Restored

**Review by caiosba — CHANGES_REQUESTED** [review]
Sawy, I just added suggestions for more clear language (ultimately this is going to be reviewed by product).

**Review by brianfleming — APPROVED** [review]
love it, thanks @melsawy 

spam

**Pull request merged**